tests/test_inference.py

```diff
@@ -1,83 +1,61 @@
-# Copyright 2021 AlQuraishi Laboratory
-#
-# Licensed under the Apache License, Version 2.0 (the "License");
-# you may not use this file except in compliance with the License.
-# You may obtain a copy of the License at
-#
-#      http://www.apache.org/licenses/LICENSE-2.0
-#
-# Unless required by applicable law or agreed to in writing, software
-# distributed under the License is distributed on an "AS IS" BASIS,
-# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
-# See the License for the specific language governing permissions and
-# limitations under the License.
-
-import time
+import os
+import copy
+import pytest
 import torch
-import ml_collections as mlc
+import pickle
+import torch.multiprocessing as mp
+from functools import partial
 
 import fastfold
 from fastfold.model.hub import AlphaFold
 from fastfold.config import model_config
 from fastfold.model.fastnn import set_chunk_size
 from fastfold.utils import inject_fastnn
-from test_data_utils import random_extra_msa_feats, random_template_feats
-from fastfold.data import data_transforms
-from fastfold.utils.tensor_utils import tensor_tree_map
+from fastfold.utils.import_weights import import_jax_weights_
+
 
+@pytest.mark.parametrize('world_size', [1, 2])
+@pytest.mark.parametrize('chunk_size', [None, 2])
+@pytest.mark.parametrize('inplace', [False, True])
+def test_state_dict(world_size, chunk_size, inplace):
+    run_func = partial(run_dist, world_size=world_size, chunk_size=chunk_size, inplace=inplace)
+    mp.spawn(run_func, nprocs=world_size)
 
-consts = mlc.ConfigDict(
-    {
-        "n_res": 11,
-        "n_seq": 13,
-        "n_templ": 3,
-        "n_extra": 17,
-    }
-)
 
-def inference():
+def run_dist(rank, world_size, chunk_size, inplace):
+    os.environ['RANK'] = str(rank)
+    os.environ['LOCAL_RANK'] = str(rank)
+    os.environ['WORLD_SIZE'] = str(world_size)
+    # init distributed for Dynamic Axial Parallelism
     fastfold.distributed.init_dap()
+    inference(chunk_size, inplace)
+    
 
-    n_seq = consts.n_seq
-    n_templ = consts.n_templ
-    n_res = consts.n_res
-    n_extra_seq = consts.n_extra
 
+def inference(chunk_size, inplace):
 
     config = model_config('model_1')
+    config.globals.chunk_size = chunk_size
+    config.globals.inplace = False
```

Comment:
Inplace is always False. You can set fastfold's inplace by model.config.globals.inplace.

Reply:
No, there is no `inplace` in config. If you don't set it, it will remind you no `config.globals.inplace`  when forward.

Reply:
Yes. We can first set inplace to False to build openfold model. And set fastfold's inplace according to the args.

Reply:
I think building a model and loading weights require more time than deepcopy. I set the `inplace=True` before fastmodel run, it is the same.

Reply:
I missed that line T_T. It looks good to me.